### PR TITLE
check existance w/o traversing symlinks

### PR DESCRIPTION
**Pull request by atahabaki**
# Description

Currently `path exists` checks the file/folder's existence by traversing symlinks. I've added a `-n` switch/flag that disables symlink traversing, similar to what `path expand -n` does.

## The Long Story (for those interested)

Hello! 👋 While working on one of my scripts, I discovered that the `path exists` command was traversing symlinks. This meant that even if the file existed, it would fail if the pointed location didn't exist. To address this, I've introduced a new `-n` flag, which I borrowed from the `path expand` command. This addition should make the behavior more consistent within the *path commands universe*.

## But, is it any useful?
 
```nushell
let compat = /run/media/userX/DriveX/steam/steamapps/compatdata
if "symlink" == ($compat | path expand -n | path type) {}
# to this
if ($compat | path exists -n) {}
```

# User-Facing Changes

Users, will not efect. Unless they use the mentioned `-n` flag/switch.

### CODE REVIEW

**Comment by WindSoilder** [comment]
Hi, thanks for your contribution!
Would you please add tests for you changes?

**Review by WindSoilder — CHANGES_REQUESTED** [review]
Hi, thanks for your contribution!
Would you please add tests for you changes?

File: crates/nu-command/src/path/exists.rs [hunk 1]
```diff
@@ -152,4 +170,26 @@ mod tests {
 
         test_examples(SubCommand {})
     }
+
+    #[test]
+    fn test_check_symlink_exists() {
```
Comment (WindSoilder) [inline crates/nu-command/src/path/exists.rs:175]: It's not a right place for adding test which required with `nu!` macro, it's better to add to `crates/nu-command/tests/commands/path/exists.rs`

File: crates/nu-command/src/path/exists.rs [hunk 2]
```diff
@@ -152,4 +170,26 @@ mod tests {
...
+            std::os::unix::fs::symlink(dirs.test().join(symlink_target), dirs.test().join(symlink))
+                .unwrap();
+            #[cfg(windows)]
+            std::os::windows::fs::symlink_file(
+                dirs.test().join(symlink_target),
+                dirs.test().join(symlink),
+            )
+            .unwrap();
+
+            let shell_res = nu!(cwd: sandbox.cwd(), "'symlink' | path exists -n");
```
Comment (WindSoilder) [inline crates/nu-command/src/path/exists.rs:191]: To test `-n` works, I think you need to remove `symlink` target first.  Then try `'symlink' | path exists -n`

**Review by WindSoilder — CHANGES_REQUESTED** [review]
Thanks for adding tests!  I think we need to change something

**Comment by fdncred** [comment]
if we can get these tests moved, we can probably land this.

File: crates/nu-command/src/path/exists.rs (hunk 2 above)
Comment (atahabaki) [reply crates/nu-command/src/path/exists.rs:191]: Sorry for late reply.

A points to B. (A -> B). `A` is `symlink`, `B` is `symlink target`.

Without the flag -n, `path exists` checks whether the A exists and then B... But what if B does not exist and just A exists? i've added `-n` to cover that.


below line just creates the A, not B. Which is fine. Why would I need to remove the A?

```
std::os::unix::fs::symlink(dirs.test().join(symlink_target), dirs.test().join(symlink)).unwrap();
```

test case checks_if_existing_file_exists() covers that situation i suppose.

**Review by WindSoilder — APPROVED** [review]
Thanks!  Looks good to me

**Pull request merged**